app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java

```diff
@@ -182,6 +183,20 @@ private void loadData() {
         }
     }
 
+    private static Map<String, String> merge(Map<String, String> existingProps, Map<String, String> newProps) {
```

Comment:
There's a `io.syndesis.common.util.CollectionsUtils::aggregate` you can use instead.

Reply:
Many tests failing ans it does not allow properties duplication: `java.lang.IllegalStateException: Duplicate key smtp` when calling the `loadData()` method.